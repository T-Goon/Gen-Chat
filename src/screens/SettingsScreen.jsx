import React, { useContext } from "react";
import { View, StyleSheet } from "react-native";

import ButtonPrim from "../components/ButtonPrim";
import { Context } from '../Context';

const SettingsScreen = ({ navigation }) => {
    const { remove } = useContext(Context);

    return (
        <View style={styles.container}>
            <ButtonPrim buttonStyles={styles.logout} onPress={async () => {
                await remove('token');
                navigation.reset({ index: 1, routes: [{ name: 'Login' }] });
            }
            }>Logout</ButtonPrim>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'white',
        paddingVertical: 20,
        paddingHorizontal: 10
    },
    logout: {
        alignSelf: 'center',
        backgroundColor: 'hsl(0, 100%, 50%)'
    }
});

export default SettingsScreen;