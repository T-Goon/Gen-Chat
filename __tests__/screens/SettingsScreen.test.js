import React from 'react';
// import renderer from 'react-test-renderer';
import { render, fireEvent, act } from '@testing-library/react-native';

import SettingsScreen from '../../src/screens/SettingsScreen';

describe('<SettingsScreen/>', () => {
    it('has i child', async () => {
        const tree = render(<SettingsScreen />).toJSON();

        expect(tree.children.length).toBe(1);
    });

    it('renders correctly', async () => {
        const tree = render(<SettingsScreen />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});