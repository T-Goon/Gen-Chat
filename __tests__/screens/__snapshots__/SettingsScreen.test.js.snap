// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen/> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "hsl(0, 100%, 50%)",
        "borderRadius": 10,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Logout
    </Text>
  </View>
</View>
`;
